File: app/src/main/java/io/github/ovso/blackbox/utils/SchedulersFacade.java
package io.github.ovso.blackbox.utils;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;
import javax.inject.Inject;

public class SchedulersFacade {

  @Inject
  public SchedulersFacade() {
  }

  /**
   * IO thread pool scheduler
   */
  public Scheduler io() {
    return Schedulers.io();
  }

  /**
   * Computation thread pool scheduler
   */
  public Scheduler computation() {
    return Schedulers.computation();
  }

  /**
   * Main Thread scheduler
   */
  public Scheduler ui() {
    return AndroidSchedulers.mainThread();
  }
}

File: app/src/main/java/io/github/ovso/blackbox/utils/SchedulersFacade.kt
package io.github.ovso.blackbox.utils

import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.core.Scheduler
import io.reactivex.rxjava3.schedulers.Schedulers
import javax.inject.Inject

class SchedulersFacade @Inject
constructor() {

  /**
   * IO thread pool scheduler
   */
  fun io(): Scheduler {
    return Schedulers.io()
  }

  /**
   * Main Thread scheduler
   */
  fun ui(): Scheduler {
    return AndroidSchedulers.mainThread()
  }
}
